packages/server-common/test/handlers/calendar/unsubscribe-calendars.test.ts

```diff
@@ -0,0 +1,190 @@
+import { Settings } from 'luxon';
+import {
+  unsubscribeCalendarsHandlerFactory,
+  UnsubscribeFromEventChanges,
+} from '../../../src';
+import { getCalendarDataObject } from '../../fixtures/calendar.fixtures';
+import {
+  createEventBridgeScheduledEventMock,
+  createHandlerContext,
+} from '../../helpers/events';
+import { calendarDataProviderMock } from '../../mocks/calendar-data-provider.mock';
+import { loggerMock as logger } from '../../mocks/logger.mock';
+
+describe('Unsubscribe calendar handler', () => {
+  const unsubscribeMock: jest.MockedFunction<UnsubscribeFromEventChanges> =
+    jest.fn();
+  const unsubscribeCalendarsHandler = unsubscribeCalendarsHandlerFactory(
+    calendarDataProviderMock,
+    unsubscribeMock,
+    logger,
+  );
+  const invokeHandler = () =>
+    unsubscribeCalendarsHandler(
+      createEventBridgeScheduledEventMock(),
+      createHandlerContext(),
+    );
+
+  const getCalendarId = (id: string) => `cms:${id}`;
+
+  const unsubscribeCalendarsHandlerWithIdFunction =
+    unsubscribeCalendarsHandlerFactory(
+      calendarDataProviderMock,
+      unsubscribeMock,
+      logger,
+      getCalendarId,
+    );
+  const invokeHandlerWithIdFunction = () =>
+    unsubscribeCalendarsHandlerWithIdFunction(
+      createEventBridgeScheduledEventMock(),
+      createHandlerContext(),
+    );
+
+  const fakeNow = 1614697798681;
+  const realDate = Date.now.bind(Date);
+
+  beforeAll(() => {
+    Settings.now = () => fakeNow;
+  });
+
+  afterAll(() => {
+    Settings.now = realDate;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  test('Should get the list of calendars expiring within 24h', async () => {
+    calendarDataProviderMock.fetch.mockResolvedValueOnce({
+      total: 0,
+      items: [],
+    });
+
+    await invokeHandler();
+
+    expect(calendarDataProviderMock.fetch).toHaveBeenCalledWith({
+      maxExpiration: fakeNow + 24 * 60 * 60 * 1000,
+    });
+  });
+
+  test('Should get the list of calendars and unsubscribe each of them', async () => {
+    const calendarDataObject1 = {
+      ...getCalendarDataObject(),
+      channelId: undefined,
+    };
+    const calendarDataObject2 = {
+      ...getCalendarDataObject(),
+      id: 'uuid2',
+      resourceId: 'resource-id-2',
+      channelId: undefined,
```

Comment:
I realise this is copied from the tests before but I was wondering if it's actually correct - we are meant to unsubscribe fully working subscriptions here right? So channelId should be present. We could examine an edge case for when it's not but again this would be an edge case. What do you think?